packages/lib/src/interactions/hooks.ts

```diff
@@ -185,21 +186,40 @@ export function useInteraction(id: string, value: InteractionEntry) {
 }
 
 export function useModifierKeyPressed(modifierKeys: ModifierKey[]): boolean {
```

Comment:
I could have gone for a `useEffect` that resets the map when `modifierKeys` changes (with `useDeepCompareEffect` to avoid having to memoise the array), but this seemed overly convoluted.

Instead:

1. I initialise a `Map` that doesn't trigger re-renders when updated.
2. Then, every time a key is pressed or released:
  a. I update the map without even bothering to check whether the key pressed/released is a modifier key.
  b. I call `checkAllPressed` which loops through the `modifierKeys` array (instead of the map) and checks whether every  given modifier key is pressed (by looking in the `pressedKeys` map).
  c. I then update the `allPressed` state, but only if needed.

This solution means that we don't re-render every time a key is pressed or released but only when the `allPressed` state changes.